packages/pwa/app/commerce-api/auth.js

```diff
@@ -49,21 +52,17 @@ class Auth {
 
         // To store tokens as cookies
         // change the next line to
-        // this._storage = new CookieStorage()
-        this._storage = new LocalStorage()
+        // this._storage = this._onClient ? new CookieStorage() : new Map()
+        this._storage = this._onClient ? new LocalStorage() : new Map()
```

Comment:
Instead of having many `this._onClient` checks in the code, we can use a `Map` on the server side.

Reply:
Open question: Do we risk exposing user data by having this stateful storage on the server?

Reply:
Great question! We instantiate a singleton of `Auth` for each request at the beginning of react rendering pipeline, in AppConfig.restore, see [here](https://github.com/SalesforceCommerceCloud/pwa-kit/blob/develop/packages/pwa/app/components/_app-config/index.jsx#L69), restore is being called the very first thing during rendering for each request, see [here](https://github.com/SalesforceCommerceCloud/pwa-kit/blob/develop/packages/pwa-kit-react-sdk/src/ssr/server/react-rendering.js#L136). So we don't leak tokens across requests.

That being said, in the future, if we make architectural changes and the Auth singleton does not re-instantiate per request, then we have a problem, but I don't see that happening any time soon.